src/test/java/com/specialwarriors/conal/GithubRepoServiceTest.java

```diff
@@ -0,0 +1,550 @@
+        ReflectionTestUtils.setField(mockRepo, "id", 1L);
+        ReflectionTestUtils.setField(mockUser, "id", 1L);
+        mockRepo.setUser(mockUser);
+
+        mockContributorList = List.of(
+                new Contributor("test@gmail.com"),
+                new Contributor("test2@gmail.com")
+        );
+
+        mockNotificationAgreements = List.of(
+                new NotificationAgreement(NotificationType.VOTE),
+                new NotificationAgreement(NotificationType.CONTRIBUTION)
+        );
+
+        mockRepo.addContributors(mockContributorList);
+        mockRepo.assignRepoIdToNotificationAgreements(mockNotificationAgreements);
+    }
+
+    @Test
+    @DisplayName("레포를 생성할 수 있다.")
+    void createRepo() {
+        // given
+        final String name = "repoName";
+        final String url = "https://github.com/owner/reponame";
+        final LocalDate endDate = LocalDate.now();
+
+        GithubRepoCreateRequest request = new GithubRepoCreateRequest(
+                name,
+                url,
+                endDate,
+                Set.of("test@test.com")
+        );
+
+        GithubRepo githubRepo = new GithubRepo(name, url, endDate);
+        GithubRepoCreateResponse response = new GithubRepoCreateResponse(
+                "owner",
+                name
+        );
+
+        given(userQuery.findById(1L)).willReturn(mockUser);
+        given(githubRepoRepository.save(any(GithubRepo.class))).willReturn(mockRepo);
+        given(githubRepoMapper.toGithubRepo(request)).willReturn(githubRepo);
+        given(contributorRepository.saveAll(anyList())).willReturn(mockContributorList);
+        given(githubRepoMapper.toGithubRepoCreateResponse(
+                any(String.class), any(String.class))
+        ).willReturn(response);
+
+        // when
+        GithubRepoCreateResponse result = githubRepoService.createGithubRepo(1L, request);
+
+        // then
+        assertThat(result.repo()).isEqualTo(name);
+        assertThat(result.owner()).isEqualTo("owner");
+    }
+
+    @Test
+    @DisplayName("레포 아이디와 유저 아이디로 깃 레포를 조회할 수 있다.")
+    void findRepoByUserIdAndRepositoryId() {
+        // given
+        given(githubRepoQuery.findByUserIdAndRepositoryId(1L, 1L)).willReturn(mockRepo);
+
+        String[] ownerAndRepo = UrlUtil.urlToOwnerAndReponame(mockRepo.getUrl());
+        given(githubRepoMapper.toGithubRepoGetResponse(mockRepo, ownerAndRepo[0], ownerAndRepo[1],
+                1L)).willReturn(
+                new GithubRepoGetResponse(
+                        mockUser.getId(),
+                        mockRepo.getId(),
+                        mockRepo.getName(),
+                        mockRepo.getUrl(),
+                        mockRepo.getEndDate(),
+                        ownerAndRepo[0],
+                        ownerAndRepo[1]
+                )
+        );
+
+        // when
+        GithubRepoGetResponse result = githubRepoService.getGithubRepoInfo(1L, 1L);
```

Comment:
앞선 리뷰와 마찬가지로 여러 번 사용되는 값의 경우 변수를 통해 의미를 명확히 하면 더 좋을 것 같습니다.
```suggestion
        GithubRepoGetResponse result = githubRepoService.getGithubRepoInfo(userId, repoId);
```

Reply:
넵 의미 명확히 하겠습니다!